test/hooks/useFetch.test.ts

```diff
@@ -0,0 +1,47 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+import { generateMockFetch } from 'test/testUtils';
+import { useFetch } from 'hooks';
+import artists from 'data/artists.json';
+import events from 'data/events.json';
+import venues from 'data/venues.json';
+import type { HakkasanResponse } from 'lib/hakkasan';
+
+describe('useFetch(endpoint)', () => {
+  test('should fetch data from the Next.js proxy API', async () => {
+    const endpoint = '/api/hakkasan?ref=events';
+    const expected = {
+      artists,
+      events,
+      venues,
+    };
+
+    global.fetch = generateMockFetch({
+      json: () => Promise.resolve(expected),
+    });
+
+    const { result } = renderHook(() => useFetch<HakkasanResponse>(endpoint));
+
+    await act(async () => {
+      await result.current.fetchData();
```

Comment:
You can pass in `result.current.fetchData` directly as an argument to `act()` here and in [line 41](https://github.com/joannerd/hakkasan/pull/3/files#diff-9544fb45add65fa9f45714818f0dc8496eae7372c61a297779fe2aebb1265c1dR41). It might be good to keep in mind that creating inline `async` arrow functions when you don't need to leads to unnecessary overhead for the TS compiler.

For example, the compiled output of this inline async arrow function looks like this:
```js
    return __awaiter(this, void 0, void 0, function* () {
        yield act(() => __awaiter(this, void 0, void 0, function* () {
            yield fetchData();
        }));
    });
```

... and passing the function directly looks like this:
```js
    return __awaiter(this, void 0, void 0, function* () {
        yield act(fetchData);
    });
```

You can see the compiler intersperses the callback through an `awaiter` wrapper when it doesn't actually need to. In other words best practice would be to avoid using `async`/`await` when there's nothing to actually `await` for.

You can see the full TS Playground example [here](https://www.typescriptlang.org/play?target=2#code/IYZwngdgxgBAZgV2gFwJYHsLwKbKgCwBFhlgAKASgC4YAFAJ3QFtURsAeAN3VQBMA+GAG8AUDHEwI2AO51GLNmXrYQ6ADadsMALyC2yACqom2dAmRlKOwctUbsZJL2xxUU3hQA0MAIwAGAIoKEQBfERFEFAwsYCgLKAAjGitdOWZWDm4+fgphMQllZAR6LETKAG5Q8NBIWEi46JhkFWQfK1EJGGBpYFRkLriyGugYFMEOzq6evpw8IhJyCkrOkKWqkWG6pAbMJpaAJnb88W7e-tiLOFwCYlI1kKA).

Reply:
Ooooh thank you!!